### ISSUE TITLE
[4.0] Argument 2 passed to Joomla\Component\Finder\Administrator\Indexer\Result::addNestedTaxonomy()

### ISSUE THREAD

**Issue by Quy** [joomla/joomla-cms#26114]
### Steps to reproduce the issue
Install `Multilingual Sample Data`
Go to Content > Categories
Unpublish `Category (en-gb)`

@Hackwar 

### Actual result
```
Argument 2 passed to Joomla\Component\Finder\Administrator\Indexer\Result::addNestedTaxonomy() must implement interface Joomla\CMS\Tree\ImmutableNodeInterface, null given, called in \joomla-cms-4.0-dev\plugins\finder\content\content.php on line 322 
```
PHP error log:
```
PHP Notice:  Trying to get property 'published' of non-object in \joomla-cms-4.0-dev\plugins\finder\content\content.php on line 322
PHP Notice:  Trying to get property 'access' of non-object in \joomla-cms-4.0-dev\plugins\finder\content\content.php on line 322
PHP Notice:  Trying to get property 'language' of non-object in \joomla-cms-4.0-dev\plugins\finder\content\content.php on line 322
```

### Additional comments
```

Call stack
--
# | Function | Location
1 | () | JROOT\administrator\components\com_finder\Indexer\Result.php:427
2 | Joomla\Component\Finder\Administrator\Indexer\Result->addNestedTaxonomy() | JROOT\plugins\finder\content\content.php:322
3 | PlgFinderContent->index() | JROOT\administrator\components\com_finder\Indexer\Adapter.php:332
4 | Joomla\Component\Finder\Administrator\Indexer\Adapter->reindex() | JROOT\administrator\components\com_finder\Indexer\Adapter.php:451
5 | Joomla\Component\Finder\Administrator\Indexer\Adapter->categoryStateChange() | JROOT\plugins\finder\content\content.php:105
6 | PlgFinderContent->onFinderCategoryChangeState() | JROOT\libraries\src\Plugin\CMSPlugin.php:285
7 | Joomla\CMS\Plugin\CMSPlugin->Joomla\CMS\Plugin\{closure}() | JROOT\libraries\vendor\joomla\event\src\Dispatcher.php:495
8 | Joomla\Event\Dispatcher->dispatch() | JROOT\libraries\src\Application\EventAware.php:111
9 | Joomla\CMS\Application\WebApplication->triggerEvent() | JROOT\plugins\content\finder\finder.php:123
10 | PlgContentFinder->onCategoryChangeState() | JROOT\libraries\src\Plugin\CMSPlugin.php:285
11 | Joomla\CMS\Plugin\CMSPlugin->Joomla\CMS\Plugin\{closure}() | JROOT\libraries\vendor\joomla\event\src\Dispatcher.php:495
12 | Joomla\Event\Dispatcher->dispatch() | JROOT\libraries\src\Application\EventAware.php:111
13 | Joomla\CMS\Application\WebApplication->triggerEvent() | JROOT\administrator\components\com_categories\Model\CategoryModel.php:752
14 | Joomla\Component\Categories\Administrator\Model\CategoryModel->publish() | JROOT\libraries\src\MVC\Controller\AdminController.php:214
15 | Joomla\CMS\MVC\Controller\AdminController->publish() | JROOT\libraries\src\MVC\Controller\BaseController.php:728
16 | Joomla\CMS\MVC\Controller\BaseController->execute() | JROOT\libraries\src\Dispatcher\ComponentDispatcher.php:146
17 | Joomla\CMS\Dispatcher\ComponentDispatcher->dispatch() | JROOT\libraries\src\Component\ComponentHelper.php:383
18 | Joomla\CMS\Component\ComponentHelper::renderComponent() | JROOT\libraries\src\Application\AdministratorApplication.php:117
19 | Joomla\CMS\Application\AdministratorApplication->dispatch() | JROOT\libraries\src\Application\AdministratorApplication.php:160
20 | Joomla\CMS\Application\AdministratorApplication->doExecute() | JROOT\libraries\src\Application\CMSApplication.php:241
21 | Joomla\CMS\Application\CMSApplication->execute() | JROOT\administrator\includes\app.php:63
22 | require_once() | JROOT\administrator\index.php:36``

**Comment by brianteeman** [comment]
Confirmed

**Comment by SharkyKZ** [comment]
Test PR #26235 please.

**Comment by ghost** [comment]
Closed as having Pull Request #26235
